src/Microsoft.AspNet.Mvc.Extensions/TypeHelper.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    internal static class TypeHelper
+    {
+        /// <summary>
+        /// Given an object, adds each instance property with a public get method as a key and its
+        /// associated value to a dictionary.
+        ///
+        /// If the object is already an <see cref="IDictionary{string, object}"/> instance, then a copy
+        /// is returned.
+        /// </summary>
+        /// <remarks>
+        /// The implementation of PropertyHelper will cache the property accessors per-type. This is
+        /// faster when the the same type is used multiple times with ObjectToDictionary.
+        /// </remarks>
+        public static IDictionary<string, object> ObjectToDictionary(object value)
```

Comment:
We should just move this to the `PropertyHelper` code since that's what it relies on.

This is a significant enough amount of code that it makes sense to have it be truly shared.